modules/7xbidBidAdapter.js

```diff
@@ -0,0 +1,157 @@
+  var hashes = url.slice(url.indexOf('?') + 1).split('&');
+  for (var i = 0; i < hashes.length; i++) {
+    hash = hashes[i].split('=');
+    myJson[hash[0]] = hash[1];
+  }
+  return myJson;
+}
+
+export const spec = {
+  code: BIDDER_CODE,
+  aliases: [BIDDER_ALIAS], // short code
+  supportedMediaTypes: SUPPORTED_MEDIA_TYPES,
+  isBidRequestValid: function(bid) {
+    if (!(bid.params.placementId)) {
+      return false;
+    }
+
+    if (bid.params.hasOwnProperty('currency') &&
+      SUPPORTED_CURRENCIES.indexOf(bid.params.currency) === -1) {
+      utils.logInfo('Invalid currency type, we support only JPY and USD!')
+      return false;
+    }
+
+    return true;
+  },
+  /**
+    * Make a server request from the list of BidRequests.
+    *
+    * @param {validBidRequests[]} - an array of bids
+    * @return ServerRequest Info describing the request to the server.
+    */
+  buildRequests: function(validBidRequests, bidderRequest) {
+    let serverRequests = [];
+    var refererInfo;
+    if (bidderRequest && bidderRequest.refererInfo) {
+      refererInfo = bidderRequest.refererInfo;
+    }
+    var g = (typeof (geparams) !== 'undefined' && typeof (geparams) == 'object' && geparams) ? geparams : {};
+    validBidRequests.forEach((bid, i) => {
+      let endpoint = ENDPOINT_BANNER
+      let data = {
+        'placementid': bid.params.placementId,
+        'cur': bid.params.hasOwnProperty('currency') ? bid.params.currency : DEFAULT_CURRENCY,
+        'ua': navigator.userAgent,
+        'adtk': _encodeURIComponent(g.lat ? '0' : '1'),
+        'loc': utils.getTopWindowUrl(),
+        'topframe': (window.parent === window.self) ? 1 : 0,
+        'sw': screen && screen.width,
+        'sh': screen && screen.height,
+        'cb': Math.floor(Math.random() * 99999999999),
+        'tpaf': 1,
+        'cks': 1,
+        'requestid': bid.bidId
+      };
+
+      if (bid.hasOwnProperty('nativeParams')) {
+        endpoint = ENDPOINT_NATIVE
+        data.tkf = 1 // return url tracker
+        data.ad_track = '1'
+        data.apiv = '1.1.0'
+      }
+
+      if (refererInfo && refererInfo.referer) {
+        data.referer = refererInfo.referer;
+      } else {
+        data.referer = '';
+      }
+
+      serverRequests.push({
+        method: 'GET',
+        url: endpoint,
+        data: utils.parseQueryStringParameters(data)
+      })
+    })
+
+    return serverRequests;
+  },
+  interpretResponse: function(serverResponse, request) {
+    const data = _getUrlVars(request.data)
+    const successBid = serverResponse.body || {};
+    let bidResponses = [];
+    if (successBid.hasOwnProperty(data.placementid)) {
+      let bid = successBid[data.placementid]
+      let bidResponse = {
+        requestId: bid.requestid,
+        cpm: bid.price,
+        creativeId: bid.creativeId,
+        currency: bid.cur,
+        netRevenue: NET_REVENUE,
+        ttl: 700
+      };
+
+      if (bid.hasOwnProperty('title')) { // it is native ad response
```

Comment:
Guessing you trust that your ad server will only respond with native if the request was native, so no check needed for the bidRequest being native here?

Reply:
only native bid response has `title` field so that I use it to check